### PR TITLE
Add User authentication to SDLB (#859)

### PR DESCRIPTION
**Pull request by zzeekk**
see #859

### CODE REVIEW

File: sdl-core/src/main/scala/io/smartdatalake/app/UIBackendConfig.scala [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+ *       useIdToken = true
+ *       clientId = "###FILE#ui-auth;clientId###"
+ *       user = "###FILE#ui-auth;user###"
+ *       password = "###FILE#ui-auth;pwd###"
+ *     }
+ *   }
+ * }}}
+ *
+ * @param baseUrl        API base URL for upload. Add tenant, repo (required) and env as query parameters, e.g. https://<host>?tenant=<tenant>&repo=<repository>&env=<environment>"
+ * @param tenant         Optional unique name for your organization within the SDLB UI. The userId set through authMode need to have access to this tenant.
```
Comment (pgruetter) [inline sdl-core/src/main/scala/io/smartdatalake/app/UIBackendConfig.scala]:
```suggestion
 * @param tenant         Optional unique name for your organization within the SDLB UI. The userId set through authMode needs to have access to this tenant.
```

File: sdl-core/src/main/scala/io/smartdatalake/app/UIBackendConfig.scala [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+ *                       Default for tenant is the special "PrivateTenant". In this case the API will use the userId of authenticated user as tenant.
```
Comment (pgruetter) [inline sdl-core/src/main/scala/io/smartdatalake/app/UIBackendConfig.scala]:
```suggestion
 *                       Default for tenant is the special "PrivateTenant". In this case the API will use the userId of the authenticated user as tenant.
```

File: sdl-core/src/main/scala/io/smartdatalake/workflow/connection/authMode/SSLCertsAuthMode.scala [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ */
+
+package io.smartdatalake.workflow.connection.authMode
+
+import com.typesafe.config.Config
+import io.smartdatalake.config.{FromConfigFactory, InstanceRegistry}
+import io.smartdatalake.util.secrets.{SecretsUtil, StringOrSecret}
+
+/**
+ * Validate by SSL Certificates : Only location an credentials. Additional attributes should be
```
Comment (pgruetter) [inline sdl-core/src/main/scala/io/smartdatalake/workflow/connection/authMode/SSLCertsAuthMode.scala]: What is meant by "Only location an credentials" ?

**Review by pgruetter — APPROVED** [review]
Works great! I see there's a few breaking changes regarding class names and removal of keycloak. Can you please take note of these changes for the release notes?

File: sdl-core/src/main/scala/io/smartdatalake/workflow/connection/authMode/SSLCertsAuthMode.scala (hunk 3 above)
Comment (zzeekk) [reply sdl-core/src/main/scala/io/smartdatalake/workflow/connection/authMode/SSLCertsAuthMode.scala]: This is not a new comment, just refactoring in a seperate class ;-)
I will improve it...

**Pull request merged**